sls-versions/src/main/java/com/palantir/slspackaging/versions/SlsProductVersions.java

```diff
@@ -6,7 +6,7 @@
  * Utility methods for checking whether version strings are valid SLS version strings.
  */
 public class SlsProductVersions {
-    private static final Pattern NON_ORDERABLE_VERSION = Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]+-[a-z0-9-.]*$");
+    private static final Pattern NON_ORDERABLE_VERSION = Pattern.compile("^[0-9]+\\.[0-9]+\\.[0-9]+((-[a-z0-9-.]*)|(\\.dirty))$");
```

Comment:
ie we want to allow `1.0.0.dirty` and `1.0.0-1-gabcdef.dirty` but did not really want to allow `1.0.0-rc.1`